### DISCUSSION TITLE
Deduct auto-creation fee from `CryptoTransfer` payer

### DISCUSSION THREAD

**Discussion by tinker-michaelj** [hiero-ledger/hiero-improvement-proposals#541] [Ideas]
[HIP-32](https://hips.hedera.com/hip/hip-32) defined a new way to create Hedera account, by submitting a `CryptoTransfer` that sends hbar to a protobuf representation of a key, which is called an _alias_.

When two conditions hold,
  1. The alias has not already been used in a such a `CryptoTransfer`; and,
  2. The sent hbar is enough to cover the account creation fee.

then the network creates a new account with the key implied by the alias. It deducts the account creation fee from the sent hbar, and uses whatever is left over as the new account's balance. 

This means auto-account creation does not work when sending only a units of a fungible token or NFT to a new alias, because a fee cannot be deducted from such quantities.

---

💡 We propose to charge auto-account creation fees to the payer of a `CryptoTransfer` that credits a new alias **anywhere** in body of a `CryptoTransfer`; whether in the hbar transfer list, or in an HTS token transfer list.

It will remain illegal to debit any asset from an uncreated alias in a `CryptoTransfer`.

**Comment by bugbytesinc** [comment]
Wait, that's a breaking change, "whether in the hbar transfer list, or...", so I can send one tinybar to an alias that has not been created and the payer will pay the account creator fee, correct?
